exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,560 @@
+#pragma once
+
+#include <sparkplug_b/sparkplug_b.pb.h>
+#include <async_mqtt/all.hpp>
+#include <boost/asio.hpp>
+
+#include <tfc/dbus/string_maker.hpp>
+
+#include "config.hpp"
+
+using org::eclipse::tahu::protobuf::Payload;
+using org::eclipse::tahu::protobuf::Payload_Metric;
+
+namespace asio = boost::asio;
+namespace details = tfc::ipc::details;
+
+class network_manager {
+public:
+  auto connect_socket(auto&& socket, auto&& resolved_ip) -> asio::awaitable<void> {
+    co_await asio::async_connect(socket, resolved_ip, asio::use_awaitable);
+  }
+
+  auto set_sni_hostname(auto&& native_handle, std::string broker_address) -> void {
+    if (!SSL_set_tlsext_host_name(native_handle, broker_address.c_str())) {
```

Comment:
The function is setting the SNI of the socket, I have not found how to do this using openssl.

Reply:
@magni-mar he means that you are not including the header the method is from.
You get it from async-mqtt but since you are using the method directly you should include the openssl header. Specifically ``` #include <openssl/ssl.h>``` I think.

Reply:
https://www.openssl.org/docs/man1.1.1/man3/SSL_set_tlsext_host_name.html

Ahh okey, I misunderstood, will include the header.